### ISSUE TITLE
Function/promise stack is not balanced when error is caught in try

### ISSUE THREAD

**Issue by kondziu** [PRL-PRG/promisedyntracer#37]
I executed the following:
```r
library(promisedyntracer)
dt <- create_dyntracer("test.sqlite", verbose=TRUE, truncate=TRUE)
dyntrace(dt, try(unknown_object))
```

Here I expect an error message in try: 
```
[1] "Error in try(unknown_stuff) : object 'unknown_stuff' not found\n"
```

But the code should execute and therefore be traced correctly. However, we get: 

```
DYNTRACE LOG - WARNING - probes.cpp · end(...) · 119 - Function/promise stack is not balanced: 5 remaining
```

This means that there is a LONGJMP somewhere which we didn't handle correctly. When a LONGJMP happens we should unwind the obsolete functions from our internal stacks. We do this for some occurances of LONGJMP, but clearly something is missing.

**Comment by kondziu** [comment]
Here's a trace: [trace.txt](https://github.com/PRL-PRG/promisedyntracer/files/1706909/trace.txt)

**Comment by kondziu** [comment]
The fix_unwinding branch as of 2e8c81cffe75fd3567cd868ed5d635177f6b7946 fixes most of these errors. In previous tests the error occured in:
- synthpop
- widals
- wrapr
- storr
- SSM
- spind
- naptime
- lightsout
- laeken
- grattah
- GGMselect
- FeaLect
- esaddle
- dismo
- cheddar
- BMS

After the fix they occur only in:
- GGMselect
- spind
- FeaLect

**Comment by kondziu** [comment]
Looks fixed in 3431e41153b58fd153c9b004790fc8879b411978.
